scipy/optimize/setup.py

```diff
@@ -73,7 +73,7 @@ def configuration(parent_package='', top_path=None):
     ext._pre_build_hook = pre_build_hook
 
     sources = ['moduleTNC.c', 'tnc.c']
-    config.add_extension('_moduleTNC',
+    config.add_extension('moduleTNC',
```

Comment:
@andyfaff I just ran into this - why did you remove the underscore? That seems like a bug to me, it's not a public submodule.

Reply:
Opened gh-15270 to revert this change.